### ISSUE TITLE
Broken DAG - how are dependencies installed correctly?

### ISSUE THREAD

**Issue by SvenLieber** [viaacode/knowledge-graph-etl#1]
Great stuff!

However, I wonder how needed Python dependencies are installed, as currently I get the error message below when executing the steps described below. Furthermore, according to the Apache Airflow documentation there seem to be several ways to provide some dependencies i.e. own Dockerfile and plugin folder. 

```
Broken DAG: [/opt/airflow/dags/load_knowledge_graph.py] Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/load_knowledge_graph.py", line 15, in <module>
    from construction_site import parse_json
ModuleNotFoundError: No module named 'construction_site'
```

If I'm understanding the Apache Airflow [documentation](https://airflow.apache.org/docs/docker-stack/build.html#example-of-adding-pypi-package) correctly, an own image has to be created where dependencies are installed in the Dockerfile. However, there is also the plugins directory. Thus I'm wondering how you installed dependencies.

## Steps to reproduce

1. Clone the repo
2. set the `AIRFLOW_UID` in `.env` to the host user's UID
3. set the `AIRFLOW_GID` in `.env` to `0`
4. Mount the correct dag directory by changing line 54 of the docker-compose file to `- ./airflow_files/dags:/opt/airflow/dags`
5. execute `docker-compose up airflow-init`
6. execute `docker-compuse up` and login via UI

**Comment by mielvds** [comment]
The `construction_site` package is missing and this is a private repo for now. This should be made public soon, I just have to clean up the repo first. But feel free to replace the `parse_json` function by a command line run of https://github.com/AtomGraph/JSON2RDF though. The `parse_json` function is a python port of JSON2RDF.

About dependencies: when you use a `PythonOperator`, all worker nodes need to have the same python environment and you have to configure this manually. Or use the `PythonVirtualenvOperator`.
The general recommendation though, is to use docker containers for all tasks in combination with the `DockerOperator`, so this DAG is not the best example for easy upscaling. See https://medium.com/bluecore-engineering/were-all-using-airflow-wrong-and-how-to-fix-it-a56f14cb0753

**Comment by SvenLieber** [comment]
Okay, thanks for the link I will have a look.
Currently I am on a small VM with *very* limited resources but I will keep the `DockerOperator` solution in mind for the future.

For now the following way to install needed functionality fixed my problem:

According to the Airflow documentation I created my own Dockerfile on top of Airflow (which I then use via `build` in `docker-compose.yaml`) installing `apache-airflow-providers-ftp` which was the dependency I actually needed. But I also installed `curl` and `wget` which I use for some preliminary testing in a preliminary (but probably not very recommended) setup using only `BashOperator`s.

```Dockerfile
FROM apache/airflow:2.1.2
USER root
RUN apt-get update  \
    && apt-get install -y --no-install-recommends \
    && apt-get install curl wget \
    && pip install apache-airflow-providers-ftp \
    && apt-get autoremove -yqq --purge \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*
USER airflow
```
